src/coprocessor/dag/batch/executors/util/scan_executor.rs

```diff
@@ -187,13 +187,18 @@ pub fn field_type_from_column_info(ci: &ColumnInfo) -> FieldType {
 }
 
 /// Checks whether the given columns info are supported.
+/// Enum and Set FieldType should also be valid as we don't need to decode these columns
 pub fn check_columns_info_supported(columns_info: &[ColumnInfo]) -> Result<()> {
     use cop_datatype::EvalType;
+    use cop_datatype::FieldTypeTp;
     use cop_datatype::FieldTypeAccessor;
     use std::convert::TryFrom;
 
     for column in columns_info {
-        box_try!(EvalType::try_from(column.tp()));
+        match column.tp() {
+            FieldTypeTp::Enum | FieldTypeTp::Set => (),
```

Comment:
Actually all field type tp can be accepted here I guess, not only Enum and Set, but also others, as long as it is not a handle column.

Reply:
What do you mean a handle column?

Reply:
A handle column is a column that has pk_is_handle = true in its column info. Handle column is always decoded by TiKV while other colums are not.

Reply:
So the actually check here should be whether the column is handle column or not ?

Reply:
Yes I think so. Currently I cannot think of some cases that this change can be wrong. We can leave it to integration tests.

Reply:
```
for column in columns_info {
        if column.has_pk_handle() {
            box_try!(EvalType::try_from(column.tp()));
        }
    }
```
How about this？ It will first do the has_pk_handle check first which is described before. Then is will do the old check to avoid missing other important change.

Reply:
The solution to passingg `Enum` and `Set ` only is too magic in my opnion.

Reply:
Yes. I would like to see this change and integration test should be able to check its correctness.